racket/src/expander/extract/simplify-defn.rkt

```diff
@@ -20,20 +21,90 @@
       (seteq)
       (set-union (car args) (apply union-all (cdr args)))))
 
-;; compute the variables that are the target of a set! in e
+; This function gets called when encountering a 'suspect letrec': some
+; letrec expression where the bound variable is used somewhere inside
+; the bound expression, and we can't proove (given the information
+; available here) that uses is safe: that the implicit mutation that
+; happens as the letrec defines the variables.
+;
+; Generally, we want to emit a warning in this case, as it's possible
+; that we stumbled on an actual bug (e.g. the bug fixed in git commit
+; b9b017261bf35cc42b341ca5d3b8f17e955416c2.) However, there are also
+; a handful of cases where the code is (obviously to humans) safe,
+; and our analysis is too primitive to reccognize it. What we do in
+; this case (see GH #5272) is to act soundly and indicate a warning,
```

Comment:
I prefer links or prose to issue numbers, just in case we ever lose this record, but other people don't care so much and I'm not a stickler for it :)